cmd/thanos/tools_bucket.go

```diff
@@ -347,6 +347,10 @@ func registerBucketWeb(app extkingpin.AppClause, objStoreConfig *extflag.PathOrC
 		router := route.New()
 		ins := extpromhttp.NewInstrumentationMiddleware(reg)
 
+		if *webExternalPrefix != "" {
+			router = router.WithPrefix("/" + strings.Trim(*webExternalPrefix, "/"))
```

Comment:
This is not what `external-prefix` is used for. This is what `route-prefix` is for. The default value of `route-prefix` is equal to `external-prefix` though. Take a look at how it is handled in `query.go`.

Reply:
The issue is that the bucket web currently does not pick up `route-prefix` through cmd flags. Therefore the router is not getting prefixed before creating the UI paths, as is happening in `query.go`. I am not sure whether bucket web should support `route-prefix` as it was not mentioned in the docs. If it should, then I could add code for picking `route-prefix` through cmd flags then proceed similar to `query.go`.